From b35798d998da3c1fc6a6e4c24dbb4a73a85a532d Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Sun, 18 May 2014 22:43:42 -0700
Subject: [PATCH 4135/5965] PD #91888. Fix HEVC VLD input HW buffer level
 calculation later than AXI bus transaction by setting HEVC streaming buffer
 hole to 256 bytes/

---
 drivers/amlogic/amports/esparser.c | 1 +
 drivers/amlogic/amports/tsdemux.c  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/amlogic/amports/esparser.c b/drivers/amlogic/amports/esparser.c
index 9bba4ce0b6bf..e8a56b48f966 100755
--- a/drivers/amlogic/amports/esparser.c
+++ b/drivers/amlogic/amports/esparser.c
@@ -353,6 +353,7 @@ s32 esparser_init(struct stream_buf_s *buf)
 
         SET_VREG_MASK(HEVC_STREAM_CONTROL, (1<<3)|(0<<4)); // set use_parser_vbuf_wp
         SET_VREG_MASK(HEVC_STREAM_CONTROL, 1); // set stream_fetch_enable
+        SET_VREG_MASK(HEVC_STREAM_FIFO_CTL, (1<<29)); // set stream_buffer_hole with 256 bytes
 
         video_data_parsed = 0;
     } else
diff --git a/drivers/amlogic/amports/tsdemux.c b/drivers/amlogic/amports/tsdemux.c
index 854fd16a8556..d22f75198679 100755
--- a/drivers/amlogic/amports/tsdemux.c
+++ b/drivers/amlogic/amports/tsdemux.c
@@ -511,6 +511,7 @@ s32 tsdemux_init(u32 vid, u32 aid, u32 sid)
 
         SET_VREG_MASK(HEVC_STREAM_CONTROL, (1<<3)|(0<<4)); // set use_parser_vbuf_wp
         SET_VREG_MASK(HEVC_STREAM_CONTROL, 1); // set stream_fetch_enable
+        SET_VREG_MASK(HEVC_STREAM_FIFO_CTL, (1<<29)); // set stream_buffer_hole with 256 bytes
     } else {
 #endif
         WRITE_MPEG_REG(PARSER_VIDEO_START_PTR,
-- 
2.19.0

